roles/sap_hana_preconfigure/tasks/SLES/configuration.yml

```diff
@@ -32,6 +32,38 @@
       ansible.builtin.debug:
         var: __sap_hana_preconfigure_fact_solution_configured
 
+- name: Set GRUB entries
+  when: __sap_hana_preconfigure_run_saptune
+  block:
+    # Reason for noqa: the regex do a check on the element before apply the
+    # changed item, this prevent a replace to an element that is already in the
+    # configuration
+    - name: Set GRUB entries # noqa no-changed-when
+      ansible.builtin.lineinfile:
+        path: /etc/default/grub
+        regexp: '^(GRUB_CMDLINE_LINUX_DEFAULT=(?!.* {{ item }}).*). *$'
+        line: "\\1 {{ item }}\""
+      register: set_grub_entries
+      with_items:
+        - "splash=silent"
+        - "mitigations=auto"
+        - "quiet"
+        - "numa_balancing=disable"
+        - "transparent_hugepage=never"
+        - "intel_idle.max_cstate=1"
+        - "processor.max_cstate=1"
+        - "audit=1"
+
+    - name: GRUB mkconfig # noqa command-instead-of-shell no-handler no-changed-when
```

Comment:
@Wabri You may have a look at https://github.com/sap-linuxlab/community.sap_install/blob/7720d7a025a8cca1c7dde23b46df8174e2fdf0fc/roles/sap_hana_preconfigure/tasks/RedHat/installation.yml#L39 for an example for explaining skipping `no-changed-when`.
For explaining multiple noqa exceptions, this one can serve as an example:
https://github.com/sap-linuxlab/community.sap_install/blob/7720d7a025a8cca1c7dde23b46df8174e2fdf0fc/roles/sap_hana_preconfigure/tasks/RedHat/assert-installation.yml#L88